src/iotHubMessageExplorer.ts

```diff
@@ -56,19 +64,105 @@ export class IoTHubMessageExplorer extends BaseExplorer {
 
         try {
             this._eventHubClient = EventHubClient.fromConnectionString(iotHubConnectionString);
+            this.updateMonitorStatus(true);
             this._outputChannel.show();
             const deviceLabel = deviceItem ? `[${deviceItem.deviceId}]` : "all devices";
             this.outputLine(Constants.IoTHubMonitorLabel, `Start monitoring D2C message for ${deviceLabel} ...`);
             TelemetryClient.sendEvent(Constants.IoTHubAIStartMonitorEvent, { deviceType: deviceItem ? deviceItem.contextValue : "" });
             this.startMonitor(Constants.IoTHubMonitorLabel, consumerGroup, deviceItem);
         } catch (e) {
+            this.updateMonitorStatus(false);
             this.outputLine(Constants.IoTHubMonitorLabel, e);
             TelemetryClient.sendEvent(Constants.IoTHubAIStartMonitorEvent, { Result: "Exception", Message: e });
         }
     }
 
     public stopMonitorIoTHubMessage(): void {
         this.stopMonitor(Constants.IoTHubMonitorLabel, Constants.IoTHubAIStopMonitorEvent);
-        this._eventHubClient = null;
+    }
+
+    private startMonitor(label: string, consumerGroup: string, deviceItem?: DeviceItem) {
+        if (this._eventHubClient) {
+            const monitorD2CBeforeNowInMinutes = Utility.getConfiguration().get<number>("monitorD2CBeforeNowInMinutes");
+            const startAfterTime = new Date(Date.now() - 1000 * 60 * monitorD2CBeforeNowInMinutes);
+            this._eventHubClient.open()

```

Comment:
does open exception catched?

Reply:
the promise exception seems not caught. The then block just called when resolved, seems there is no chained catch

Reply:
Most are receiver error. And it is caught by `receiver.on("errorReceived", this.printError(this._outputChannel, label));`
Since I just copy the whole code of `startMonitor` from baseExplorer.ts, I create a seperate task for code refactor: https://github.com/Microsoft/vscode-azure-iot-toolkit/issues/232

Reply:
ok